src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -1717,6 +1718,38 @@ public String getVariableNamePattern() {
 					})
 					.serializer(new EnumSerializer<>(Cat.Type.class)));
 		}
+		
+		EnumUtils<Attribute> attributes = new EnumUtils<>(Attribute.class, "attribute types");
+		Classes.registerClass(new ClassInfo<>(Attribute.class, "attributetype")
+				.user("attribute ?types?")
+				.name("Attribute Type")
+				.description("The attribute type of an entity. Note that this type does not contain any numerical values."
+						+ "See <a href='https://minecraft.gamepedia.com/Attribute#Attributes'>attribute types</a> for more info.")
+				.defaultExpression(new EventValueExpression<>(Attribute.class))
+				.examples(attributes.getAllNames())
+				.since("INSERT VERSION")
+				.parser(new Parser<Attribute>() {
+					@Override
+					@Nullable
+					public Attribute parse(String input, ParseContext context) {
+						return attributes.parse(input);
+					}
+					
+					@Override
+					public String toString(Attribute a, int flags) {
+						return attributes.toString(a, flags);
+					}
+					
+					@Override
+					public String toVariableNameString(Attribute a) {
+						return a.toString();
+					}
+					
+					@Override
+					public String getVariableNamePattern() {
+						return "\\S+";
```

Comment:
after the variable string form is changed, this is not true as it may contain a space

Reply:
Oh right, since attributes toString can contain spaces. Sorry, I'm not too familiar with regex, what should be the proper pattern for this?

Reply:
it would be `[\sA-Za-z]+` for a name containing letters or spaces